### ISSUE TITLE
Safe defaults for agent file logging

### ISSUE THREAD

**Issue by schmichael** [hashicorp/nomad#6603]
I missed a couple things when reviewing #6429:

1. It would save users some calculator usage to allow them to specify the max file size in MB instead of bytes. That would also match our jobspec: https://www.nomadproject.io/docs/job-specification/logs.html#max_file_size
2. Defaulting to an unbounded file size (0) seems unsafe. Let's default to 10 files of 10 MB. That way it keeps the storage overhead of Nomad more predictable and easier to factor into the [`client.reserved`](https://www.nomadproject.io/docs/configuration/client.html#reserved-parameters) stanza.

As a stretch goal we could probably also now implement this as an alternate Sink on hclogger once #6499 lands. That would allow logging at a different log level as well so you could log to stdout (and therefore to your central aggregator) at INFO, but have 100MB worth of DEBUG logs locally in case something went wrong. Probably not worth worrying about in the 0.10.x timeline. Maybe we'll even get this for free once Consul implements hclogger.

**Comment by endocrimes** [comment]
Currently we have the same defaults as Consul, which potentially has some value for someone running both (most nomad users I would assume).

We _might_ want to work with consul to improve this across both.

**Comment by schmichael** [comment]
Consul does not appear to support configuring file logging rotation from the config file -- or at least it's not documented that I see: https://www.consul.io/docs/agent/options.html#log_file

So by only supporting via the config file we're already diverging from Consul. I think safe defaults are worth diverging in behavior -- especially when it makes us more internally consistent with our jobspec logging behavior.

**Comment by schmichael** [comment]
Going to close since we're shipping 0.10.2 with it the way it is. We can reopen based on user demand/experience.
